apps/prairielearn/src/lib/grading.ts

```diff
@@ -187,7 +187,8 @@ export async function saveSubmission(
 ): Promise<{ submission_id: string; variant: Variant }> {
   const submission: Partial<Submission> & SubmissionDataForSaving = {
     ...submissionData,
-    raw_submitted_answer: submissionData.submitted_answer,
+    // This is cloned to avoid double-storing workspace files.
+    raw_submitted_answer: structuredClone(submissionData.submitted_answer),
```

Comment:
Note that for complex questions that store large/deep complex data this will increase memory usage somewhat since we'll now have a copy of that data. However, the bulk of the data in this object is usually strings, and I believe v8 reuses them in `structuredClone` since they're immutable.

Reply:
To avoid this issue, would it be viable to make the clone in line 210, when we actually know that the files will be added? Note that the vast majority of questions that have workspaces don't have other elements, so in these questions `raw_submitted_answer` will usually be very small.

Reply:
Yeah, that's reasonable! Changed in 237414c.